packages/format-library/src/link/inline.js

```diff
@@ -91,6 +77,35 @@ const LinkViewerUrl = ( { url } ) => {
 	);
 };
 
+const URLPopoverAtLink = ( { isActive, addingLink, value, ...props } ) => {
+	const anchorRect = useMemo( () => {
+		const range = window.getSelection().getRangeAt( 0 );
+		if ( ! range ) {
+			return;
+		}
+
+		if ( addingLink ) {
+			return getRectangleFromRange( range );
+		}
+
+		let element = range.startContainer;
+		while ( element.nodeType !== window.Node.ELEMENT_NODE ) {
+			element = element.parentNode;
+		}
+
+		const closest = element.closest( 'a' );
+		if ( closest ) {
+			return closest.getBoundingClientRect();
+		}
+	}, [ isActive, addingLink, value.start, value.end ] );
+
+	if ( ! anchorRect ) {
+		return null;
+	}
+
+	return <URLPopover getAnchorRect={ () => anchorRect } { ...props } />;
```

Comment:
I wonder if passing `anchorRect` directly to `URLPopover` could prevent the delays: `Popover` could simply check object equality? The calculation already happened anyway, so the `setInterval` doesn't really make much sense if there's no calculation needed to be done.

Reply:
> I wonder if passing `anchorRect` directly to `URLPopover` could prevent the delays: `Popover` could simply check object equality? The calculation already happened anyway, so the `setInterval` doesn't really make much sense if there's no calculation needed to be done.

Yeah, I think that could work. It would require some updates to the `Popover` component, but they seem relatively small. I'll give it a shot.

Reply:
This works great! Much smoother. :)